### DISCUSSION TITLE
interactively changed Kconfig settings affects .config in build directory that is not intended for adding to git repo

### DISCUSSION THREAD

**Discussion by rjmaris** [zephyrproject-rtos/zephyr#61743] [General]
As a (very) former user of uclinux.dist, I'm wondering about the fact that the .config file that is merged (e.g. with prj.conf & generated via menuconfig is placed in a subdirectory of the build directory (I'm talking about a project that is created as a sibling of the zephyr directory, according to a T2 topology).

I know that project specific config is stored in `prj.conf`, but many detailconfigurations are managed conveniently via menuconfig/guiconfig and moreover, the result of specific user settings should anyhow be part of my project versioning.
In uclinux.dist, this is done via a copy of **toplevel config**, **dist-level config** and **linux config** in vendors/xxx/  (at the moment I was not able to reconstruct the process part of copying back actual changes via menuconfig to the /vensors/xxx/ directory - which is versioned).

Of course this workflow does not apply in my actual case because the own project is not part of the standard zephyr repository. But I don't find hints in `doc/build/Kconfig`. Of course, I won't manually transfer changes through menuconfig over to `prj.conf`.
Do I miss something which is documented? Just I found that upon building with pristine the modified .config is removed (as expected)...

**Comment by thedjnK** [comment]
From menuconfig, press `d` and you will be prompted to save a defconfig file, this is a minified `prj.conf` compatible file, you can move the options over.

**Comment by rjmaris** [comment]
I just tried this. In menuconfig the file defconfig is created, but menuconfig hangs. In guiconfig no hang-up. However, this file is by default stored in the `build` subtree...
Moreover: with or without this file, whenever I change prj.conf, the .config output is saved with its default content...

Sidenote: Zephyr version is 3.2.99.
